### PR TITLE
Test flaky devfile modification test

### LINKED ISSUE #6601: Flaky test: `modifying memoryLimit for container in Devfile`
/kind bug
/area testing

Examples:
- https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-6586-openshift-tests-4060.txt

```
Timed out after 180.001s.
  Expected
      <string>:  ⚠  Pod is Terminating
       ✗  Finished executing the application (command: devrun) [40s]
       ⚠  No pod exists
       ⚠  Pod is Pending
       ✓  Pod is Running
       •  Syncing files into the container  ...
      
 ✓  Syncing files into the container [382ms]
       •  Building your application in container (command: devbuild)  ...
      
 ✓  Building your application in container (command: devbuild) [6s]
       •  Executing the application (command: devrun)  ...
       -  Forwarding from 127.0.0.1:42585 -> 3000
      
      ↪ Dev mode
       Keyboard Commands:
      [Ctrl+c] - Exit and delete resources from the cluster
           [p] - Manually apply local changes to the application on the cluster
      
  To satisfy at least one of these matchers: [%!s(*matchers.ContainSubstringMatcher=&{Pushing files... []}) %!s(*matchers.ContainSubstringMatcher=&{Updating Component... []})]
  In [BeforeEach] at: /go/odo_1/tests/helper/helper_run.go:64
------------------------------
Summarizing 1 Failure:
  [FAIL] odo dev command tests port-forwarding for the component when devfile has single endpoint when running odo dev when modifying memoryLimit for container in Devfile [BeforeEach] should react on the Devfile modification
  /go/odo_1/tests/helper/helper_run.go:64

Ran 421 of 771 Specs in 1277.997 seconds
FAIL! -- 420 Passed | 1 Failed | 0 Pending | 350 Skipped
```

_Originally posted by @rm3l in https://github.com/redhat-developer/odo/issues/6586#issuecomment-1432996001_

### PR DESCRIPTION
**Pull request by valaparthvi**
**What type of PR is this:**
/area testing
/kind bug

**What does this PR do / why we need it:**

**Which issue(s) this PR fixes:**

Fixes #6601 

**PR acceptance criteria:**

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

File: tests/integration/cmd_dev_test.go [hunk 1]
```diff
@@ -1038,6 +1039,20 @@ ComponentSettings:
 								var stdout string
 								var stderr string
 								BeforeEach(func() {
+									var (
```
Comment (feloy) [inline tests/integration/cmd_dev_test.go:1042]: Youcould move the Skip here:

```
if manual {
  if os.Getenv("SKIP_KEY_PRESS") == "true"
```

**Comment by valaparthvi** [comment]
```sh

  [FAILED] user-guides/quickstart/docs-mdx/nodejs/nodejs_odo_dev_output.mdx
  Expected
      <string>: Â Â (
      Â Â 	"""
      Â Â 	... // 9 identical lines
      Â Â 	â€¢  Waiting for Kubernetes resources  ...
      Â Â 	âš   Pod is Pending
      -Â 	âœ“  Pod is Running
      Â Â 	âœ“  Syncing files into the container [1s]
      Â Â 	âœ“  Building your application in container (command: install) [1s]
      Â Â 	... // 10 identical lines
      Â Â 	[Ctrl+c] - Exit and delete resources from the cluster
      Â Â 	[p] - Manually apply local changes to the application on the cluster
      +Â 	âœ“  Pod is Running
      Â Â 	```
      Â Â 	"""
      Â Â )
      
  to be empty
  In [It] at: /go/odo_1/tests/documentation/user-guides/doc_user_guides_quickstart_test.go:104 @ 04/25/23 12:18:22.625
------------------------------
Spawning '/go/bin/odo init' from /tmp/3354671968
```
/override Kubernetes-Integration-Tests/Kubernetes-Docs-Integration-Tests

**Comment by feloy** [comment]
LGTM

**Comment by feloy** [comment]
/override windows-integration-test/Windows-test

Tests pass previously

```
  [oc] Unable to connect to the server: dial tcp 158.177.79.34:31700: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
```

**Comment by feloy** [comment]
Thanks @valaparthvi for this fix

**Comment by valaparthvi** [comment]
@feloy can you approve this PR?

**Comment by feloy** [comment]
/lgtm

Sorry, I thought I approved it when overriding the tests

**Pull request merged**